### ISSUE TITLE
Lint JavaScript within HTML

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#116]

**Comment by vegerot** [comment]
As a minimal starting spot, will this treat every `script` tag as an independent source of code, or will this feature include the absurd HTML spec regarding how script tags interact.  And will QLJS lint script tag attributes, or will that be left to QLHTML?

**Comment by strager** [comment]
> As a minimal starting spot, will this treat every `script` tag as an independent source of code, or will this feature include the absurd HTML spec regarding how script tags interact?

Those are good questions. I haven't thought about it. We can't reasonably peer into `<script src="https://...">` and see what global variables are declared for future scripts.

I guess we'd need to disable some checks (such as knowing the set of declared global variables), but otherwise, we can do all syntax checking and many semantic checks. Then we can lint the script blocks independently (maybe **iN** pAr**AlL**eL).

> And will QLJS lint script tag attributes

Yes.

**Comment by strager** [comment]
This might help development: https://htmlparser.info/

**Comment by strager** [comment]
A nice lint rule:
```html
<form onsubmit="publish"> ... </form>
```
warning: missing '()' after function in event handler
